pkg/util/csr.go

```diff
@@ -85,3 +88,18 @@ func RenewTimeFromNotAfter(notBefore time.Time, notAfter time.Time, renewBeforeS
 
 	return dur, nil
 }
+
+func KeyUsagesFromAttributes(attr map[string]string) []cmapi.KeyUsage {
+	usageCSV := attr[csiapi.KeyUsagesKey]
+
+	if len(usageCSV) == 0 {
+		return nil
+	}
+
+	var keyUsages []cmapi.KeyUsage
+	for _, usage := range strings.Split(usageCSV, ",") {
+		keyUsages = append(keyUsages, cmapi.KeyUsage(usage))
+	}
```

Comment:
Same comment as above....did you consider stripping surrounding space from each usage?
And I don't now if the comma separated values might ever be quoted? If so you could use the csv package: https://golang.org/pkg/encoding/csv/#example_Reader

And should we de-duplicate the usages here?
You could use the kubernetes `sets` package.

Reply:
I'm not too keen on doing things like removing duplicates and leaving that to the user. There is no confusion then.

Will remove the white space though